### ISSUE TITLE
508-defect-2 [SCREENREADER, KEYBOARD]: Complementary information should be contained in an aside

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#18694]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

[This is similar to the design system ticket #6379](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6379)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Josh_

## Details

Currently, "Herramientas útiles en línea" (Helpful online tools) and "Más información" (More information) are contained within `<section>`s inside the `<main>` . Semantically, since these sections are complimentary, they should be contained within an `<aside>`. By doing so, they will be listed as a separate region for screen reader users to navigate to and more discoverable.

However, nesting the `<section>`s within an `<aside>` within the `<main>` will cause AXE to throw a best practice suggestion. "Embedding an `<aside>` element in another landmark may disable screen reader functionality allowing users to navigate through complementary content." As such, we recommend moving the `<aside>` _outside_ of the `<main>`. 

Note here: there may be layout and styling issues when moving the `<aside>` outside of the `<main>`. If the solution listed is not possible, there are several other alternatives we can discuss.

## Acceptance Criteria

- [ ] "Herramientas útiles en línea" and "Más información" are nested within an `<aside>`
- [ ] The `<aside>` is _outside_ of `<main>`

## Environment

Tested on Safari/VO, Chrome/JAWS, Firefox/NVDA

## Steps to Recreate

1. Enter `https://www.va.gov/coronavirus-veteran-frequently-asked-questions-esp/` in browser
2. Inspect "Herramientas útiles en línea", confirm it is within a `<section>` nested under `<main>`

## Proposed Solution 

After testing 4 different methods of addressing this on the browser/at combinations listed in environment, I believe the best path forward is to:
1. contain both `<section>`s within an `<aside>`
2. move the `<aside>` outside of the `<main>`

This will add _one_ `complementary` region to all browser/AT combinations. This is intentional, as landmarks are meant to denote significance. Having more than one `complementary` region by making the `<section>`s into separate `<article>`s or by adding `aria-labelledby` to the sections will only add unnecessary clutter to regional navigation.

### Before

```html
<main>
  <!-- Content before complementary info --> 
  <section data-template="paragraphs/list_of_link_teasers" data-entity-id="27858" class="field_content_block">
    <h2 id="herramientas-útiles-en-línea" class="va-nav-linkslist-heading">
       Herramientas útiles en línea
    </h2>
  </section>
  <section data-template="paragraphs/list_of_link_teasers" data-entity-id="27862" class="field_content_block">
      <h2 id="más-información" class="va-nav-linkslist-heading">
          Más información
      </h2>
  </section>
</main>
```

### After

```html
<main>
  <!-- Content before complementary info --> 
</main>

<!-- Add an aside below main and move the sections -->
<aside>
  <section data-template="paragraphs/list_of_link_teasers" data-entity-id="27858" class="field_content_block">
    <h2 id="herramientas-útiles-en-línea" class="va-nav-linkslist-heading">
       Herramientas útiles en línea
    </h2>
  </section>
  <section data-template="paragraphs/list_of_link_teasers" data-entity-id="27862" class="field_content_block">
      <h2 id="más-información" class="va-nav-linkslist-heading">
          Más información
      </h2>
  </section>
</aside
```

## WCAG or Vendor Guidance (optional)

* [Complementary landmarks and/or asides are top level](https://dequeuniversity.com/rules/axe/3.5/landmark-complementary-is-top-level?application=coconut)
* [W3C ARIA Landmarks](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/complementary.html)

## Screenshots or Trace Logs

<img width="1368" alt="Screen Shot 2021-01-18 at 10 55 54 AM" src="https://user-images.githubusercontent.com/14154792/104963571-d3ef8680-59a8-11eb-8de9-54c093fc0cde.png">

**Comment by joshkimux** [comment]
`<aside>`s usually have a light grey background to block them off as separate from the primary content, so that may want to be considered as well!

**Comment by DanielleThierryUSDSVA** [comment]
@SKasimow @youngfreezyVA Just a note on this one as well that this isn't specific to this page, but is rather a result of a Drupal component used sitewide.

**Comment by SKasimow** [comment]
@youngfreezyVA will check to see if there is a site-wide solution being developed for this.

**Comment by youngfreezyVA** [comment]
i searched for "Aside" in all open issues
https://github.com/department-of-veterans-affairs/va.gov-team/issues?q=is%3Aissue+is%3Aopen+aside
and found this:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/6379

**Comment by SKasimow** [comment]
@youngfreezyVA good detective work! But yes, unclear where/who this landed with (Trevor?).  I'll see if @brianalloyd has any more insight on this issue. 

cc: @DanielleThierryUSDSVA

**Comment by joshkimux** [comment]
@SKasimow @youngfreezyVA 
Here's another ticket for additional context https://github.com/department-of-veterans-affairs/va.gov-team/issues/3556
Brian's team would be a great resource to reach out to on this.

**Comment by brianalloyd** [comment]
@SKasimow the [#6379](https://github.com/department-of-veterans-affairs/va.gov-team/issues/6379) was back in March 2020 which was before my time and plan to program this into our grooming session for tomorrow in hopes to add more context.  I'll update this thread in support once we've reviewed as a team.

**Comment by brianalloyd** [comment]
Grooming discussion: Gist: Calling for a change to side element on CV page would be calling for change across the site and not unique to just this page.  This would result in a great deal of template changes, also would result in a high risk venture.  Needs dedicated developer focus time to work this issue.

**Comment by laflannery** [comment]
Digging into this, this is not longer clear-cut guidance and the aside doesn't need to be top level. There are instances where the <aside> doesn't necessarily need to be a top-level landmark, and this would be one of those cases. Its also important to note that axe is not flagging this as an issue.

Below are 2 resources I have found to support this markup and the ongoing discussion of the new guidance:


- [MDN documentation](https://developer.mozilla.org/en-US/docs/Learn/Accessibility/HTML#page_layouts) with example page layout with nested aside
- https://github.com/w3c/aria-practices/issues/1469 regarding how the guidance on whether or not aside must be top level or not
